### DISCUSSION TITLE
Tiling Shell v12.0 | Snap Assistant and auto-tile buttons on window menu, touch screen support and more!

### DISCUSSION THREAD

**Discussion by domferr** [domferr/tilingshell#93] [Announcements]
- **New**: Right-click on a window to use the auto-tile buttons and the snap assistant from there! Closes #74 and #77. The auto-tile buttons allow you to tile to the leftmost or rightmost empty tile, based on your selected layout. The "Move to best tile" button suggests tiling to the nearest empty tile to the center of the screen. This is something new in the GNOME ecosystem, especially considering how the layout and tile buttons are drawn: feel free to share your thoughts and ideas by opening a new issue!

<p align="center"><img src="https://github.com/user-attachments/assets/d660779a-7549-4858-b149-59edad076483" width=650/></p>

<p align="center">Check the demonstration video to see it in action!</p>

[Screencast from 2024-07-13 18-21-57.webm](https://github.com/user-attachments/assets/8fd79faa-a476-4b55-b79c-6329e4b59519)

> The original idea was to show the snap assistant when hovering the maximize button (as it is done on Windows 11). Unfortunately, GNOME doesn't let us handle the hovering of the maximize button or add another button near it...

- **New**: keybindings for moving/tile windows now consider other monitors too, so you can move and tile the window from one monitor to another. Closes #73 
- **New**: auto-resize now works when resizing with window-action key. Closes #76 
- **Fix**: Windows no longer lose their position after OS suspension. Closes #85
- **Fix**: Keybindings for moving windows accross the tiles are more precise
- **Fix**: Snap assistant, with blur enabled, had wrong colors on light themes (left picture). Now it has the right transparency and colors (right picture)
<p align="center">
<img src="https://github.com/domferr/tilingshell/assets/14203981/53037692-f58d-4c64-af29-04518d52a341" width=360 />
<img src="https://github.com/domferr/tilingshell/assets/14203981/faf2c94d-8c55-4a53-b547-3a801ab0a013" width=260/>
</p>

- **Fix**: in some edge-cases the indicator was showing the monitors with a wrong order. Closes #92

**Comment by domferr** [comment]
:star: This update of Tiling Shell is landing soon on GSE too! It is currently under review

**Reply by efogdev** [reply]
Any chance to get a dev build? I have a bug but I'm not sure if it is still relevant.

Edit: just seen the you've updated the release files. I've tried it and the bug is still relevant.
I'm going to record a video with my phone again because I want to show the keyboard and I didn't find any software to show keyboard events on the screen that works with Wayland. Are you ok with that?

**Reply by domferr** [reply]
Thank you @EfogDev! I think that's a great idea. Feel free to open a new issue

**Comment by domferr** [comment]
:star: This update of Tiling Shell is now on GSE too!
<div align="center">
  <a href="https://extensions.gnome.org/extension/7065/tiling-shell/" >
      <img src="https://img.shields.io/badge/Install%20from-extensions.gnome.org-4A86CF?style=for-the-badge&logo=Gnome&logoColor=white"/>
  </a>
</div>
